src/Microsoft.IdentityModel.Protocols.WsFederation/Configuration/WsFederationConfigurationValidator.cs

```diff
@@ -147,7 +147,11 @@ public ConfigurationValidationResult Validate(WsFederationConfiguration configur
 
                     try
                     {
+#if NET9_0_OR_GREATER
+                        cert = X509CertificateLoader.LoadCertificate(Convert.FromBase64String(signatureCertData.Current));
+#else
                         cert = new X509Certificate2(Convert.FromBase64String(signatureCertData.Current));
+#endif
```

Comment:
Since these lines repeat throughout, one code duplication reduction to consider is extract this into a helper method, so that `#if# flag is only in one place.

```csharp
cert = CertificateHelper.Load(Convert.FromBase64String(signatureCertData.Current)
```
```csharp
internal static X509Certificate2 Load(byte[] data)
{
#if NET9_0_OR_GREATER
                        return X509CertificateLoader.LoadCertificate(Convert.FromBase64String(signatureCertData.Current));
#else
                        return new X509Certificate2(Convert.FromBase64String(signatureCertData.Current));
#endif
}
```

Reply:
Great point, added a `X509CertificateHelper`.